### DISCUSSION TITLE
Feature Request: Cloudflare Warp over Speedify

### DISCUSSION THREAD

**Discussion by tiernano** [SmoothWAN/SmoothWAN#174] [General & Help]
Not sure how possible this is, but would it be possible to have Cloudflare WARP (https://1.1.1.1) running on SmoothWAN and use that for sending traffic over Speedify? I know its an extra layer, but Cloudflare would have better-peering connections than Speedify, and I think I would trust Cloudflare's privacy policy more than Speedify... the WGCF tool (https://github.com/ViRb3/wgcf) allows you to generate a Wireguard config to connect to Cloudflare...

**Comment by TalalMash** [comment]
Check this out: https://github.com/SmoothWAN/SmoothWAN/discussions/79#discussioncomment-4131043
You can use any VPN with PBR enabled.
You can also use split-tunneling, e.g set WG to a BYP routing table, and create a rule for LAN with a range e.g 172.17.17.0/24 for all clients. More: https://smoothwan.com/wanbyp/#split-tunneling-for-leak-proof-bypass

May add next release with automation as a LuCI app

**Comment by tiernano** [comment]
ok.... might be a stupid question, but how do I setup a wireguard VPN connection? When I click on "Status" I see "Wireguard" but its empty... and there doesn't seem to be a "Wireguard" option under "VPN"... I must be missing something simple... Thanks...

**Comment by tiernano** [comment]
right... i was being stupid... i created a connection though the interface, and now have wireguard setup... I still cant figure out the PBR stuff: 
* I enabled PBR
*  under advanced config, added the wg connection to the supported interfaces
*  enabled "show chain", "show enabled" and "show protocol" columns in the web interface.
* created a policy named cloudflare, local address "172.17.17.0/24", local ports 0-65535, destination "empty" (0.0.0.0/0 fails...) remote ports 0-65535, protocol ALL, chain "PREROUTING".
* save config
* start service
* still showing as old IP from Speedify...
Did I miss something?

Thanks.

**Comment by TalalMash** [comment]
Wrote a guide here, setup manually instead of PBR, and disable PBR since it seems to be unstable in later versions (?):
https://smoothwan.com/vpn/

**Comment by tiernano** [comment]
Right... i must be doing something wrong, cause I followed those steps and still getting the old IP... Do you have screen shots of the "ipv4 rules" and "static ipv4 routes" screens? Its going to be something simple, I know, but I just cant figure out what I am doing wrong... Thnaks

**Comment by TalalMash** [comment]
![image](https://user-images.githubusercontent.com/96490382/230797639-9ba0b049-df43-4b7d-b68c-ddd90591f964.png)
![image](https://user-images.githubusercontent.com/96490382/230797665-e8c30313-4dd4-48bb-b2b1-d1422c5d9e5a.png)
![image](https://user-images.githubusercontent.com/96490382/230797668-91891a5c-b8ab-4351-b332-df6708b83c5d.png)
![image](https://user-images.githubusercontent.com/96490382/230797685-9122aba1-3dea-40c1-a761-18949bd8adcc.png)
Are you using the latest version? Any custom configuration?

**Comment by tiernano** [comment]
I just checked and all the config is the same... the only difference I have is I used BYP9 (swapped it to BYP1 and no difference) and MTU on WARP was set to 1280, per the WGCF config (removed that, but still no difference). No other config differences... running on a Flint, the latest install, and IPs on the internet are Static... weird... I will try a reinstall of SmoothWAN tomorrow and see if there is a weird config issue somewhere else I am missing... ill get back to you then.

**Comment by tiernano** [comment]
Right, rebuilt my router, and walked through all steps again. I missed 1 part, when creating the VPN, under Peers, "route allowed IPs" was not on. I turned it on, walked through the rest of the steps, and all is mostly working... I get a Cloudflare IP! Problem is, it does not seem to be sending data over Speedify, so I seem to be connecting over the non-bonded links... Any ideas what I am doing wrong again?

**Comment by TalalMash** [comment]
Thanks for testing, I did notice that as well, it happens to choose WANs on first up, and setting `netifd` to bind to `connectify0` did not work for some reason, WireGuard doesn't follow the default routing table on OpenWrt.
I've added extra instructions in the meantime using firewall markings method:
https://smoothwan.com/vpn/ - *WireGuard extra steps for v0.99.9*

I've also noticed that WARP is throttled behind data center IP address like Speedify, need to do more testing to confirm.

**Comment by TalalMash** [comment]
You can also try in terminal: `ip rule add iif lo sport 2408 lookup main` (replace 2408 with your WG peer port)
Then restart the WG interface in LuCI.
